### PR TITLE
Added new validation for human turn and cleaned up code

### PR DESCRIPTION
**Pull request by krithigasr**
Has the following code changes
* Removed display of computer deck
* Show what card the computer dropped to the stack- better readability
* Added Warning tags and check if the card dropped by the human actually belongs to him
* Removed printf statements(those that are needed for testing alone)

### CODE REVIEW

File: src/game.c [hunk 1]
```diff
@@ -203,6 +207,7 @@ int computer_take_turn(void)
     else
     {   /*If there is playable card, then remove the first playable card from on hand cards list*/
         Deck_t* discard_card = remove_first_playable_card(&g_players[COMPUTER].cards_on_hand);
+        printf("Computer Drops..(%s, %s)\n", CARD_COLOR_STRING[discard_card->card.color], CARD_NAME_STRING[discard_card->card.name]);
         memcpy(&g_card_on_table, &discard_card->card, sizeof(Card_t));
         //printf("Card on the table now: (%s, %s)\n", CARD_COLOR_STRING[discard_card->card.color], CARD_NAME_STRING[discard_card->card.name]);
```
Comment (MiaYWei) [inline src/game.c]: Let's delete this printf.

File: src/cards_management.c [hunk 2]
```diff
@@ -326,7 +326,7 @@ Deck_t *remove_first_playable_card(Deck_t **pp_head)
         prev = *pp_head;             // Get reference of head node
         *pp_head = (*pp_head)->next; // Adjust head node link
         //free(prev);            // Delete prev since it contains reference to head node
-        //printf("Successfully deleted the first playable card (%s, %s) at beginning. \n", CARD_COLOR_STRING[removed_card.color], CARD_NAME_STRING[removed_card.name]);
+        //printf("Successfully deleted the first palyable card (%s, %s) at beginning. \n", CARD_COLOR_STRING[removed_card.color], CARD_NAME_STRING[removed_card.name]);
```
Comment (MiaYWei) [inline src/cards_management.c:329]: Let's delete the above all the comment out lines.

File: src/cards_management.c [hunk 3]
```diff
@@ -340,7 +340,7 @@ Deck_t *remove_first_playable_card(Deck_t **pp_head)
                 prev->next = cur->next; // Adjust links for previous node
             }
             //free(cur);                  // Delete current node
-            printf("Successfully deleted the first playable card (%s, %s) in the middle. \n", CARD_COLOR_STRING[cur->card.color], CARD_NAME_STRING[cur->card.name]);
+            //printf("Successfully deleted the first palyable card (%s, %s) in the middle. \n", CARD_COLOR_STRING[cur->card.color], CARD_NAME_STRING[cur->card.name]);
```
Comment (MiaYWei) [inline src/cards_management.c:343]: Let's delete the above all the comment out lines.

**Review by MiaYWei — APPROVED** [review]
Let's delete all the comment out lines.

File: src/game.c (hunk 1 above)
Comment (krithigasr) [reply src/game.c]: This printf is needed to make the game interactive. Personally, I felt lost having to compare the card on table to its previous value to check if the computer has dropped a card or not. I feel this make things more clear

File: src/cards_management.c (hunk 2 above)
Comment (krithigasr) [reply src/cards_management.c:329]: We'll delete if after release 2.

File: src/cards_management.c (hunk 3 above)
Comment (krithigasr) [reply src/cards_management.c:343]: We'll delete if after release 2.

File: src/game.c (hunk 1 above)
Comment (MiaYWei) [reply src/game.c]: I mean this one: //printf("Card on the table now: (%s, %s)\n", CARD_COLOR_STRING[discard_card->card.color], CARD_NAME_STRING[discard_card->card.name]);

Comment (krithigasr) [reply src/game.c]: Got it.

**Review by HongBo713 — APPROVED** [review]

**Review by aditibhardwaj98 — APPROVED** [review]

**Pull request merged**